molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/service/impl/AlgorithmServiceImpl.java

```diff
@@ -101,23 +101,27 @@ public void autoGenerateAlgorithm(EntityType sourceEntityType, EntityType target
 	public Iterable<AlgorithmEvaluation> applyAlgorithm(Attribute targetAttribute, String algorithm,
 			Iterable<Entity> sourceEntities)
 	{
-		return Iterables.transform(sourceEntities, entity ->
+		return stream(sourceEntities.spliterator(), false).map(entity ->
 		{
 			AlgorithmEvaluation algorithmResult = new AlgorithmEvaluation(entity);
-
 			Object derivedValue;
+
 			try
 			{
 				Object result = jsMagmaScriptEvaluator.eval(algorithm, entity);
```

Comment:
could you not check for null / empty instead of catching the exception ? If we expect null it does not seem like a exceptional state.